src/main/kotlin/org/rust/ide/hints/RsInlayParameterHintsProvider.kt

```diff
@@ -50,6 +52,14 @@ enum class HintType(desc: String, enabled: Boolean) {
             } else {
                 0
             }
+            if (smart && declaration?.valueParameters?.size ?: 0 == 1) {
+                if (declaration?.name == declaration?.valueParameters?.getOrNull(0)?.pat?.text) {
```

Comment:
I found the exact rules that Java uses. There's `methodName.contains(paramName)`. I think it makes sense and covers a bit more cases. We could do the same.